cmd/kubeadm/app/util/pkiutil/pki_helpers.go

```diff
@@ -556,32 +625,39 @@ func NewPrivateKey(keyType x509.PublicKeyAlgorithm) (crypto.Signer, error) {
 }
 
 // NewSignedCert creates a signed certificate using the given CA certificate and key
-func NewSignedCert(cfg *CertConfig, key crypto.Signer, caCert *x509.Certificate, caKey crypto.Signer) (*x509.Certificate, error) {
+func NewSignedCert(cfg *CertConfig, key crypto.Signer, caCert *x509.Certificate, caKey crypto.Signer, isCA bool) (*x509.Certificate, error) {
 	serial, err := cryptorand.Int(cryptorand.Reader, new(big.Int).SetInt64(math.MaxInt64))
 	if err != nil {
 		return nil, err
 	}
 	if len(cfg.CommonName) == 0 {
 		return nil, errors.New("must specify a CommonName")
 	}
-	if len(cfg.Usages) == 0 {
+	if !isCA && len(cfg.Usages) == 0 {
```

Comment:
i wonder if we should just remove the mandatory `len(cfg.Usages) > 0` here?
i believe currently we are passing either `x509.ExtKeyUsageClientAuth` or `x509.ExtKeyUsageServerAuth` to this function via config.

Reply:
The config is passed through from `NewCertAndKey` so it's probably cleaner if I just perform this check there instead. I'll push a commit for this, let me know if you prefer that version otherwise I'm happy to remove it entirely.

Reply:
> The config is passed through from NewCertAndKey so it's probably cleaner if I just perform this check there instead.

agreed.